pkg/nsx/services/subnetport/subnetport.go

```diff
@@ -373,3 +375,61 @@ func (service *SubnetPortService) Cleanup(ctx context.Context) error {
 	}
 	return nil
 }
+
+func (service *SubnetPortService) AllocatePortFromSubnet(subnet *model.VpcSubnet) bool {
```

Comment:
It looks you missed the case of restart, in which we should construct the `PortCountInfo` with the existing Subnets ?

Reply:
In the case of restarting, we do not have dirty port under creation so dirtyCount is safe to initialize as 0. And the totalIP will be calculated in this AllocatePortFromSubnet function as there is no PortCountInfo in the map.